Convert data track to MODE1/2352 format and add tool to generate checksums

sftwninja opened the pull request:
RE: #72 

### What this changes
Principally, this adds a step which takes in the ISO with the data track, and rather than appending it directly as a MODE1/2048 track, it first converts it into a MODE1/2352 track then later regenerates the EDC values.

## What's the difference between 2048 and 2352?
The best way to describe the 2352 mode is that it's 2048 mode in a neat wrapper. There are real benefits to this outside of just resolving #72, including having error correction CD burners will utilize rather than its own (explicit vs implicit EDC), and honestly my first PR should have done this but I was a little afraid of pushing such a big PR all at once.

## How it works
For each sector:
* A sync pattern is written. (12 bytes)
* The header is written. Minutes Seconds Frame and MODE1. (4 bytes)
* The data track is written. (2048 bytes)
* Finally, the EDC, or Error Detection Code is written. This is non-trivial so I didn't want this project to own this task, and instead pass it off to a tried-and-true tool I've used for a couple years now, EDCRE. We write 288 zero'd bytes, and EDCRE comes in at the end and calculates the proper EDC and ECC w/ Reed-Solomon encoding. (EDC 4 bytes, 8 zero bytes, 276 ECC bytes)

Additional note: We have to correct the EDC/ECC information. Emulators such as Mednafen/Beetle Saturn refuse to boot a game that has invalid sector checksums. Real Saturns sometimes verify the checksums, and sometimes don't, and sometimes verify it sometimes.

## Why this change matters
MODE1/2048 is a "cooked" format. It's a weird format in that _most_ software that deal with CDs attempt to convert it internally to MODE1/2352. ODEs, including ones that aren't for the Saturn, also tend to get a little funky with it. MODE1/2352, _the_ "raw" format, ensures that we send what we want to be read in the format we want with the layout we want. It's literally the layout that goes on the disc.

What I was noticing while tracking this issue down is that different systems saw the MODE1/2048 data differently. This was causing track drift.

How I confirmed the changes:
* Analyze in CDmage
* Burn, re-rip, and compare in CDmage again
* Check audio in Mednafen
* Debug disc reads in Kronos
* Verify it didn't crash Ymir
* Several burned discs and file swaps on the real Saturn

## New tool `EDCRE`
I mentioned it above, but this includes a tool called `edcre`. It can be found [here](https://github.com/alex-free/edcre). It's a tool built out of a portion of `cdrdao` that rebuilds EDC/ECC portions. I've used it in the past with Dreamcast games and I've seen it used a few times in the PSX/PS2 community when doing asset conversions.

**This adds a new component to the setup_compiler.bat script.** It now fetches EDCRE as a step after the compiler and extracts it to a directory under `tools/bin/PLATFORM`. I swap between Windows, WSL, and Linux in a VM commonly, so I wanted something that could track them independently.

## Cons and callouts
* The track conversion task is slow. I should roll the EDCRE functionality directly into a dedicated app to do this conversion in totality.
* I have not tested this on Mac OS.
* I've tested this on physical CD as well as with Fenrir. I don't own a Satiator, and testing that would be appreciated.

ReyeMe commented:
I have tested in satiator and it works correctly. The build takes quite a while now, maybe we should introduce some fast just iso/cue/wav build version for rapid developement in the future

Also setting the chmod +x in a task is not necesary, it can be do with a git command `git update-index --chmod=+x` or `git add --chmod=+x`, so please revert that

bimmerlabs reviewed (approved):
seems to be working correctly - tested with Satiator, Medafen and a CD-R.

It would be nice to have a "test build" option for prototyping (cue/iso/wav maybe), and a "release build" that creates the cue/bin, since it takes a lot longer to build.

sftwninja commented:
I extracted a portion of another bigger app I have for CD mastering, added some parallelization to it, and pushed it up to github here: https://github.com/sftwninja/iso2raw

When I'm back this evening, I'll convert the slow bash/dd script that's running so slow with the one above. The above one can process a full/entire CD _usually_ in 1-2 seconds on decent hardware, so the juice is worth the squeeze for that swap-out. This will also remove the EDCRE dependency since the above embeds its functionality.

ReyeMe commented:
I will merge you PR for now, great work 👍 . You can do further improvements in another PR, to keep things focused.

ReyeMe commented:
fix by: [sftwninja](https://github.com/sftwninja)

**Change:**
Principally, this adds a step which takes in the ISO with the data track, and rather than appending it directly as a MODE1/2048 track, it first converts it into a MODE1/2352 track then later regenerates the EDC values.
